statsmodels/tools/validation/validation.py

```diff
@@ -0,0 +1,178 @@
+from collections.abc import Mapping
+
+import numpy as np
+
+
+def bool_like(value, name, optional=False, strict=False):
+    """
+    Convert to bool or raise if not bool_like
+
+    Parameters
+    ----------
+    value : object
+        Value to verify
+    name : str
+        Variable name for exceptions
+    optional : bool
+        Flag indicating whether None is allowed
+    strict : bool
+        If True, then only allow bool. If False, allow types that support
+        casting to bool.
+
+    Returns
+    -------
+    converted : bool
+        value converted to a bool
+    """
+    if optional and value is None:
+        return value
+    extra_text = ' or None' if optional else ''
+    if strict:
+        if isinstance(value, bool):
+            return value
+        else:
+            raise TypeError('{0} must be a bool{1}'.format(name, extra_text))
+
+    if hasattr(value, 'squeeze') and callable(value.squeeze):
+        value = value.squeeze()
+    try:
+        return bool(value)
+    except Exception:
+        raise TypeError('{0} must be a bool (or bool-compatible)'
+                        '{1}'.format(name, extra_text))
+
+
+def int_like(value, name, optional=False, strict=False):
+    """
+    Convert to int or raise if not int_like
+
+    Parameters
+    ----------
+    value : object
+        Value to verify
+    name : str
+        Variable name for exceptions
+    optional : bool
+        Flag indicating whether None is allowed
+    strict : bool
+        If True, then only allow int or np.integer that are not bool. If False,
+        allow types that support integer division by 1 and conversion to int.
+
+    Returns
+    -------
+    converted : int
+        value converted to a int
+    """
+    if optional and value is None:
+        return None
+    is_bool = isinstance(value, bool)
+
+    if hasattr(value, 'squeeze') and callable(value.squeeze):
+        value = value.squeeze()
+
+    if isinstance(value, (int, np.integer)) and not is_bool:
```

Comment:
Probably should exclude np.timedelta64, which is a subclass of np.integer

Reply:
Thanks.